subprojects/ide-native/src/main/java/org/gradle/ide/xcode/plugins/XcodePlugin.java

```diff
@@ -0,0 +1,247 @@
+            workspaceFileTask.setXcodeWorkspace(xcode.getWorkspace());
+            workspaceFileTask.setOutputFile(new File(xcodeWorkspacePackageDir, "contents.xcworkspacedata"));
+            getLifecycleTask().dependsOn(workspaceFileTask);
+
+            GenerateWorkspaceSettingsFileTask workspaceSettingsFileTask = project.getTasks().create("xcodeWorkspaceWorkspaceSettings", GenerateWorkspaceSettingsFileTask.class);
+            workspaceSettingsFileTask.setOutputFile(new File(xcodeWorkspacePackageDir, "xcshareddata/WorkspaceSettings.xcsettings"));
+            getLifecycleTask().dependsOn(workspaceSettingsFileTask);
+
+            for (final Project p : project.getAllprojects()) {
+                p.getPlugins().matching(Specs.union(HAS_XCODE_PLUGIN, HAS_SUPPORTED_LANGUAGE_PLUGIN)).all(new Action<Plugin>() {
+                    @Override
+                    public void execute(Plugin plugin) {
+                        if (hasSupportedLanguagePluginFor(p) && p.getPlugins().hasPlugin(XcodePlugin.class)) {
+                            xcode.getWorkspace().getProjects().add((DefaultXcodeProject) xcodeModelFor(p).getProject());
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    private static boolean hasSupportedLanguagePluginFor(Project project) {
+        return project.getPlugins().hasPlugin(SwiftExecutablePlugin.class) || project.getPlugins().hasPlugin(SwiftModulePlugin.class);
+    }
+
+    private static XcodeExtension xcodeModelFor(Project project) {
+        return project.getExtensions().getByType(XcodeExtension.class);
+    }
+
+    private void configureForSwiftPlugin(final Project project) {
+        project.getPlugins().withType(SwiftExecutablePlugin.class, new Action<SwiftExecutablePlugin>() {
+            @Override
+            public void execute(SwiftExecutablePlugin swiftExecutablePlugin) {
+                configureXcodeForSwift(project, PBXTarget.ProductType.TOOL);
+            }
+        });
+
+        project.getPlugins().withType(SwiftModulePlugin.class, new Action<SwiftModulePlugin>() {
+            @Override
+            public void execute(SwiftModulePlugin swiftModulePlugin) {
+                configureXcodeForSwift(project, PBXTarget.ProductType.DYNAMIC_LIBRARY);
+            }
+        });
+    }
+
+    private void configureXcodeForSwift(Project project, PBXTarget.ProductType productType) {
+        if (project.getBuildFile().exists()) {
+            xcode.getProject().getSources().add(project.getBuildFile());
+        }
+
+        ConfigurableFileTree sourceTree = project.fileTree("src/main/swift");
+        sourceTree.include("**/*.swift");
+        xcode.getProject().getSources().addAll(sourceTree.getFiles());
+
+        XcodeTarget target = newTarget(project.getPath(), productType, toGradleCommand(project.getRootProject()), project.getTasks().getByName("linkMain").getPath(), project.file("build/exe/" + project.getName()), sourceTree.getFiles());
```

Comment:
This should eventually reuse the logic that determines the name and output of the link task that the 'swift-executable' or 'swift-module' plugin uses, rather than assuming this hard-coded value. Perhaps add a TODO here for now.